app/src/main/java/space/taran/arknavigator/mvp/model/repo/preview/PreviewGenerators.kt

```diff
@@ -45,26 +47,64 @@ object PreviewGenerators {
             Log.d(PREVIEWS, "Thumbnail for image $path generated in $time1 ms")
             return
         }
+
         val ext = extension(path)
         generatorsByExt[ext]?.let { generator ->
             val time2 = measureTimeMillis {
-                val thumbnail = if (ext == "txt") {
-                    resizePreviewToThumbnail(generator(path))
-                } else {
-                    val preview = generator(path)
-                    storePreview(previewPath, preview)
-                    resizePreviewToThumbnail(preview)
-                }
+                val preview = generator(path)
+                storePreview(previewPath, preview)
+                val thumbnail = resizePreviewToThumbnail(preview)
                 storeThumbnail(thumbnailPath, thumbnail)
             }
             Log.d(
                 PREVIEWS,
                 "Preview and thumbnail generated for $path in $time2 ms"
             )
+            return
```

Comment:
I think we should do better dispatch based on extension and Tika. Don't copypaste the code. The code below can be removed if you check necessary conditions before and call necessary sections of code just 1 time.

Reply:
Resolved.